src/net/neighbors.rs

```diff
@@ -3390,6 +3394,103 @@ mod test {
         })
     }
 
+    #[test]
+    #[ignore]
+    fn test_step_walk_1_neighbor_bad_epoch() {
+        with_timeout(600, || {
+            let mut peer_1_config = TestPeerConfig::from_port(31998);
+            let mut peer_2_config = TestPeerConfig::from_port(31990);
+
+            peer_1_config.connection_opts.walk_retry_count = 10;
+            peer_2_config.connection_opts.walk_retry_count = 10;
+            peer_1_config.connection_opts.walk_interval = 1;
+            peer_2_config.connection_opts.walk_interval = 1;
+
+            // peer 1 thinks its always epoch 2.0
+            peer_1_config.peer_version = 0x18000000;
```

Comment:
what happens if the 2.0 node hasn't upgraded? Will this be 0x0?

Reply:
The peer version `0x18000000` *is* the 2.0 peer version.  The 2.05 one is `0x18000005`.